### ISSUE TITLE
Possible bug in FISMrmseRecommender.java

### ISSUE THREAD

**Issue by akYoung** [guoguibing/librec#179]
For the model to train, the X  is the sum of latent vectors of relevant items except from the one of the training sample. However, the value of last training sample is retained to current training sample in current code. It seems line 101  ` X=new DenseVector(numFactors);` should be put to 134, since line 147 is adding to current X

**Comment by guoguibing** [comment]
I think you are possibly right. Let me further check this issue.

**Comment by SunYatong** [comment]
@akYoung 
Thank you for the feedback, I have fixed the bug and formatted the code,

**Comment by akYoung** [comment]
Ok, I still have a question about the training loss calculation. I find it an incremental approach in the training. Could you please tell me why it is done so? Intuitively, the loss can be calculated after the end of each iteration. But I can not easily understand the rationality behind the way in the script. Thanks ahead. @SunYatong

**Comment by yingquliang** [comment]
Sorry，you are right, because I made a mistake. Thanks for your reply.


 
From: akYoung
Date: 2017-07-29 14:21
To: guoguibing/librec
CC: Subscribed
Subject: Re: [guoguibing/librec] Possible bug in FISMrmseRecommender.java (#179)
Ok, I still have a question about the training loss calculation. I find it an incremental approach in the training. Could you please tell me why it is done so? Intuitively, the loss can be calculated after the end of each iteration. But I can not easily understand the rationality behind the way in the script. Thanks ahead. @SunYatong

**Comment by akYoung** [comment]
@yingquliang  You are welcome. Hope to see the change in the code. I will close the issue.

**Comment by SunYatong** [comment]
@akYoung 
It would cost more time to calculate loss in a separated loop after each iteration. So we calculate the loss during training just for efficiency. The calculated loss is actually the last iteration's loss.
